scripts/ci/build.sh

```diff
@@ -86,7 +86,7 @@ cat >$testsrc_dir/setup.py <<EOL
 
 from setuptools import setup
 
-VERSION = "1.0.0.dev$version"
+VERSION = "1.0.0.post1"
```

Comment:
Wouldn't `1.0.1.devN` be more feasible?

Reply:
Reverted this change for CI (I don't think it matters what version to use in CI as it will not be released.), but refactored to put `dev` inside `$version` and for edge built python wheel `$version` is changed to `postN`.

With the increasing number of unexpected hot fix release, it may not be a good idea to bump version right after a release and use `devN` for edge build versions.

Reply:
> With the increasing number of unexpected hot fix release, it may not be a good idea to bump version right after a release and use `devN` for edge build versions.

I think hotfix PRs will target `release` branch, which still has the last version, like `2.16.0`, even though the `dev` branch is now at `2.17.0`.

Reply:
Let's discuss about this in our next release meeting. The CI in `azure-cli-extensions` repo is broken now, let's merge this PR to unblock it first. If we decided to use `dev` version, we can start to do it after the next release.